### PR TITLE
[NETBEANS-4039] Added SPI for Preferred Color Profile Changes

### PR DESCRIPTION
**Pull request by lkishalmi**
A non-hack implementation of #2031

### CODE REVIEW

File: platform/core.windows/apichanges.xml [hunk 1]
```diff
@@ -84,6 +84,26 @@ is the proper place.
 
   <changes>
 
+    <change id="preferred-color-profile-spi" >
+        <api name="winsys"/>
+        <summary>Add preference for sorting editor tabs</summary>
```
Comment (neilcsmith-net) [inline platform/core.windows/apichanges.xml]: Wrong summary copy-pasted in?

Comment (lkishalmi) [reply platform/core.windows/apichanges.xml]: Yes. Copy+paste issue, fixing.

Comment (neilcsmith-net) [reply platform/core.windows/apichanges.xml]: :+1: I think this now has preferred spelt wrong though?

Comment (lkishalmi) [reply platform/core.windows/apichanges.xml]: Yes it was. Thanks!

Comment (neilcsmith-net) [reply platform/core.windows/apichanges.xml]: Thanks!  Haven't tested yet, but use of SPI over reflection hacks is a good move.

File: platform/core.windows/nbproject/project.xml [hunk 2]
```diff
@@ -211,8 +211,10 @@
             <friend-packages>
                 <friend>org.netbeans.core.multitabs</friend>
                 <friend>org.netbeans.core.nativeaccess</friend>
+                <friend>org.netbeans.modules.options.editor</friend>
```
Comment (jtulach) [inline platform/core.windows/nbproject/project.xml]: A chance to remove `friend` packages and turn the API into `public` one?

File: platform/core.windows/apichanges.xml [hunk 3]
```diff
@@ -84,6 +84,26 @@ is the proper place.
 
   <changes>
 
+    <change id="preferred-color-profile-spi" >
+        <api name="winsys"/>
+        <summary>Add SPI support for preferred color profile change.</summary>
+        <version major="2" minor="91"/>
```
Comment (jtulach) [inline platform/core.windows/apichanges.xml]: I would expect an increment of the module number in `manifest.mf`...

Comment (lkishalmi) [reply platform/core.windows/apichanges.xml]: Done. Also changed friend packages to public packages.

**Comment by lkishalmi** [comment]
Considered writing an unit test, though I'd have to mock too many things to make it meaningful, or just my unit testing craft is too week.

**Pull request merged**